app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,31 @@
+class CommentsController < ApplicationController
+ 
+  helper_method :book
+
+  def index
+    @comments = book.comments.all
+
+  end
+
+  def show
+    @comments = book.comments.all
+  end
+
+  def new  
+  end
+
+  def create
+    @comment = book.comments.new(comment_params)
+    @comment.save
```

Comment:
You can do this in one line using `create` instead of `new`, as create will return the object and save it to the database 

http://guides.rubyonrails.org/active_record_basics.html#create